### DISCUSSION TITLE
Can We Use the Component Approach to IaaC in Modernisation Platform?

### DISCUSSION THREAD

**Discussion by julialawrence** [ministryofjustice/analytical-platform#4976] [Firebreak Summer 24]
## **Idea Description** 
Mechanism for enabling terraform component approach to development in Modernisation Platform

## **Why Should We Do This** 
Analytical Platform is developing a lot of infrastructure in Modernisation Platform at the moment, focussed mainly in the Compute Account. Single location for source code for all environments means that we are frequently in conflict with each other. Adopting component-based structure like we have in AP repo would aid in that.

One off-the-wall idea is to develop a terraform module for bootstrapping new components including state and provider configuration piggy-backing on the scripts that stub out new environments in this script: https://github.com/ministryofjustice/modernisation-platform/blob/main/scripts/git-create-environments.sh

## **Definition of Done**
Develop a process that would work within the constraints of MP to use separate components. 

What skills do we need?
- AWS
- Modernisation Platform
- Terraform

**Comment by jacobwoffenden** [comment]
<img src="https://media0.giphy.com/media/3oFzlV2pq0nnFntAZ2/giphy.gif"/>
